Mathlib/Topology/Constructible.lean

```diff
@@ -0,0 +1,345 @@
+/-
+Copyright (c) 2024 Yaël Dillies. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies
+-/
+import Mathlib.Order.BooleanSubalgebra
+import Mathlib.Topology.Spectral.Hom
+
+/-!
+# Constructible sets
+
+This file defines constructible sets, which are morally sets in a topological space which we can
+make out of finite unions and intersections of open and closed sets.
+
+Precisely, constructible sets are the boolean subalgebra generated by open retro-compact sets,
+where a set is retro-compact if its intersection with every compact open set is compact.
+
+Constructible sets are useful because the image of a constructible set under a finitely presented
+morphism of schemes is a constructible set (and this is *not* true at the level of varieties).
+
+## Main declarations
+
+* `IsRetroCompact`: Predicate for a set to be retro-compact, namely to have its intersection with
+  every compact open is compact.
+* `IsConstructible`: Predicate for a set to be constructible, namely to belong to the boolean
```

Comment:
I feel that this word is too ambiguous to be exempt from a namespace. Is `Set.IsConstructible` appropriate? Or `TopologicalSpace.IsConstructible`?

Reply:
I know it's ambiguous, but I don't know which I should disambiguate it. Would `IsConstructibleSet` be fine?

Reply:
That is sufficient to disambiguate from [constructible numbers](https://en.wikipedia.org/wiki/Constructible_number), but there is also the [set-theoretic notion](https://en.wikipedia.org/wiki/Constructible_universe#Constructible_sets_are_definable_from_the_ordinals) (which might be called ZFSet.IsConstructible if it ever makes into mathlib). That's why I suggest something about topology in the name.

Reply:
Ended up putting `IsConstructible` and `IsLocallyConstructible` (but not `IsRetrocompact`) under the `Topology` namespace